src/core/server/yitam-tools.ts

```diff
@@ -5,15 +5,88 @@ import { StdioServerTransport } from "@modelcontextprotocol/sdk/server/stdio.js"
 import {
   CallToolRequestSchema,
   ListToolsRequestSchema,
+  JSONRPCMessage,
+  JSONRPCRequest,
+  JSONRPCNotification,
+  JSONRPCResponse,
+  ErrorCode,
 } from "@modelcontextprotocol/sdk/types.js";
 import dotenv from 'dotenv';
 import { DatabaseService } from '../../services/database/database-service';
 import { FormattedResult } from '../../types/qdrant';
 import { YitamTool, RetrievalConfig, RetrievalArgs } from '../../types/declarations/retrieval';
+import * as http from 'http';
+import * as crypto from 'crypto';
+import * as url from 'url';
+import * as fs from 'fs';
+import * as path from 'path';
+import { EventEmitter } from 'events';
+import { Readable } from 'stream';
 
 // Load environment variables
 dotenv.config();
 
+// Constants
```

Comment:
why we code many stuffs ourselves, please investigate that what we can use from @modelcontextprotocol SDK and refactor code to move some common code into other modules